awx/api/serializers.py

```diff
@@ -2801,6 +2802,44 @@ def create(self, validated_data):
         return credential
 
 
+class CredentialInputSourceSerializer(BaseSerializer):
+    source_credential_name = serializers.SerializerMethodField(
+        read_only=True,
+        help_text=_('The name of the source credential.')
+    )
+    source_credential_type = serializers.SerializerMethodField(
+        read_only=True,
+        help_text=_('The credential type of the source credential.')
+    )
+
+    class Meta:
+        model = CredentialInputSource
+        fields = (
+            'id',
+            'type',
+            'url',
+            'input_field_name',
+            'metadata',
+            'target_credential',
+            'source_credential',
+            'source_credential_type',
+            'source_credential_name',
+            'created',
+            'modified',
+        )
+        extra_kwargs = {
+            'input_field_name': {'required': True},
+            'target_credential': {'required': True},
+            'source_credential': {'required': True},
+        }
+
+    def get_source_credential_name(self, obj):
+        return obj.source_credential.name
```

Comment:
these 2 linked credentials don't show up in summary fields?

I'm worried that people might make subsequent requests to show them in summary fields.

Reply:
@jakemcdermott I think I agree with this; it's kind of abnormal for us to add a new _field_ like this specifically for e.g., `source_credential_name`.  This is the type of data that we *generally* put in `summary_fields`.  We should probably have a `summary_fields` on this endpoint that shows ID, name, maybe some other fields for the source and target credential.

What do you think?

Reply:
works for me